### ISSUE TITLE
Update joplin from 1.0.201.exe to 1.0.214.exe produces "fatal error"

### ISSUE THREAD

**Issue by amatter** [laurent22/joplin#3250]
Hello

I just updated a fully working Joplin-Setup-1.0.201.exe on Windows 10 to 1.0.204. After the updated finished and I tried to restart there is an error message.
![joplin_1-0-204_error_message after _update](https://user-images.githubusercontent.com/3266376/82675030-76057600-9c44-11ea-9d37-60237059ac57.png)
After Clicking ok (on the error) the message disappears and nothing happens.

Going back to the old version is not possible => similar error message

PS: I use Joplin as well on ubuntu 20.04 and PinebookPro (Manjaro) on the 1.0.201 version

thanks a lot 
Regards, Adrian

## Environment

Joplin version:  1.0.214 after update from 1.0.201
Platform: Windows 10 pro 1909
OS specifics: 

## Steps to reproduce

1.  just restart joplin

### log.txt
2020-05-22 15:37:49: "Operations completed: "
2020-05-22 15:37:49: "createRemote: 8"
2020-05-22 15:37:49: "Total folders: 12"
2020-05-22 15:37:49: "Total notes: 446"
2020-05-22 15:37:49: "Total resources: 1800"
2020-05-22 15:39:49: "Scheduling sync operation...", "0"
2020-05-22 15:39:49: "Preparing scheduled sync"
2020-05-22 15:39:49: "Starting scheduled sync"
2020-05-22 15:39:49: "TaskQueue.stop: syncDownload: waiting for tasks to complete: 0"
2020-05-22 15:39:49: "TaskQueue.stop: syncDownload: Done, waited for 0"
2020-05-22 15:39:50: "BasicDelta: Report: {"timestamp":1589919412124,"older":3621,"newer":20,"equal":1}"
2020-05-22 15:39:50: "Operations completed: "
2020-05-22 15:39:50: "fetchingTotal: 20"
2020-05-22 15:39:50: "fetchingProcessed: 20"
2020-05-22 15:39:50: "Total folders: 12"
2020-05-22 15:39:50: "Total notes: 446"
2020-05-22 15:39:50: "Total resources: 1800"
2020-05-22 15:39:51: "DecryptionWorker: cannot start because no master key is currently loaded."
2020-05-22 15:39:51: "Scheduling sync operation...", "0"
2020-05-22 15:39:51: "Preparing scheduled sync"
2020-05-22 15:39:51: "Starting scheduled sync"
2020-05-22 15:39:51: "TaskQueue.stop: syncDownload: waiting for tasks to complete: 0"
2020-05-22 15:39:51: "TaskQueue.stop: syncDownload: Done, waited for 0"
2020-05-22 15:39:52: "BasicDelta: Report: {"timestamp":1590154669880,"older":3641,"newer":0,"equal":1}"
2020-05-22 15:39:52: "Operations completed: "
2020-05-22 15:39:52: "fetchingTotal: -"
2020-05-22 15:39:52: "Total folders: 12"
2020-05-22 15:39:52: "Total notes: 446"
2020-05-22 15:39:52: "Total resources: 1800"
2020-05-22 15:39:53: "DecryptionWorker: cannot start because no master key is currently loaded."
2020-05-22 15:40:37: "Profile directory: C:/Users/matter/.config/joplin-desktop"
2020-05-22 15:40:58: "Profile directory: C:/Users/matter/.config/joplin-desktop"
2020-05-22 15:41:26: "Profile directory: C:/Users/matter/.config/joplin-desktop"
2020-05-22 15:42:08: "Profile directory: C:/Users/matter/.config/joplin-desktop"
2020-05-22 15:44:27: "Profile directory: C:/Users/matter/.config/joplin-desktop"
2020-05-22 16:02:39: "Profile directory: C:/Users/matter/.config/joplin-desktop"

### log-autoupdate.txt:
2020-05-19 20:52:41: "checkForUpdates: Done."
2020-05-22 15:34:55: "checkForUpdates: Initializing..."
2020-05-22 15:34:55: "checkForUpdates: Checking with options {"includePreReleases":false}"
2020-05-22 15:34:55: "Current version: 1.0.201"
2020-05-22 15:34:55: "Latest version: 1.0.214"
2020-05-22 15:34:55: "Is Pre-release:", "false"
2020-05-22 15:38:37: "checkForUpdates: Done."
2020-05-22 15:39:18: "checkForUpdates: Starting..."
2020-05-22 15:39:18: "checkForUpdates: Checking with options {"includePreReleases":false}"
2020-05-22 15:39:18: "Current version: 1.0.201"
2020-05-22 15:39:18: "Latest version: 1.0.214"
2020-05-22 15:39:18: "Is Pre-release:", "false"
2020-05-22 15:39:19: "checkForUpdates: Done."

**Comment by Rolograaf** [comment]
Close Joplin completely with right mouse button in taskbar, I have this everytime. Or restart Windows, it will probably be solved

**Comment by laurent22** [comment]
@Rolograaf, I've never seen this error before. Do you mean it can happen when trying to open the app while it's still running in the tray icon?

**Comment by amatter** [comment]
Hello, 
Thank you. I just followed @Rolograaf 's hint.
1. checked for joplin process in taskmanager => nothing. Start Joplin => same error message as in screenshot above
2. Did a Windows restart. Start Joplin => same error message as in screenshot above 
So the Problem is still there. Therefor I avoid an upgrade on my other systems, as long there is no solution/workaround.

**Comment by laurent22** [comment]
Unfortunately your error message means that the database got somehow corrupted. Do you have a recent backup? As that would be the best option to restore everything. Otherwise, you can try to open the profile database in SQLite Browser and run the query `SELECT * FROM sqlite_master;` then maybe I can see what's wrong with the db and tell you how to fix it.

**Comment by amatter** [comment]
@laurent22 thank you.
While joplin was not running I copied the c:\Users\matter\.config\joplin-desktop\database.sqlite from another functional joplin 1.0.201 instance (on win 10 pro as well) from my notebook to my desktop. 
The instances are synced by nextcloud client.
After starting joplin 1.0.214 on desktop I got exactly the same error window.

log-database.txt (trying to start with copied 1.0.201 db)
```
2020-05-23 11:18:56: "Database was open successfully"
2020-05-23 11:18:56: "Checking for database schema update..."
2020-05-23 11:18:56: "Current database version", "28"
2020-05-23 11:18:56: "Converting database to version 29"
2020-05-23 11:18:56: "BEGIN TRANSACTION"
2020-05-23 11:18:56: "ALTER TABLE version ADD COLUMN table_fields_version INT NOT NULL DEFAULT 0"
2020-05-23 11:18:56: "UPDATE version SET version = ?"
2020-05-23 11:18:56: "[29]"
2020-05-23 11:18:56: "COMMIT"
2020-05-23 11:18:56: "Initializing tables..."
2020-05-23 11:18:56: "PRAGMA table_info("folders")"
2020-05-23 11:18:56: "PRAGMA table_info("notes")"
2020-05-23 11:18:56: "PRAGMA table_info("tags")"
2020-05-23 11:18:56: "PRAGMA table_info("note_tags")"
2020-05-23 11:18:56: "PRAGMA table_info("sync_items")"
2020-05-23 11:18:56: "PRAGMA table_info("version")"
2020-05-23 11:18:56: "PRAGMA table_info("deleted_items")"
2020-05-23 11:18:56: "PRAGMA table_info("settings")"
2020-05-23 11:18:56: "PRAGMA table_info("alarms")"
2020-05-23 11:18:56: "PRAGMA table_info("master_keys")"
2020-05-23 11:18:56: "PRAGMA table_info("item_changes")"
2020-05-23 11:18:56: "PRAGMA table_info("note_resources")"
2020-05-23 11:18:56: "PRAGMA table_info("resource_local_states")"
2020-05-23 11:18:56: "PRAGMA table_info("resources")"
2020-05-23 11:18:56: "PRAGMA table_info("notes_normalized")"
2020-05-23 11:18:56: "PRAGMA table_info("revisions")"
2020-05-23 11:18:56: "PRAGMA table_info("migrations")"
2020-05-23 11:18:56: "PRAGMA table_info("resources_to_download")"
2020-05-23 11:18:56: "PRAGMA table_info("key_values")"
2020-05-23 11:18:56: "PRAGMA table_info("sqlite_stat1")" 
```
which shows no error in Log but the the fatal error in separate window.

Using again the original db file (1.0.214 ):
```
2020-05-23 11:21:34: "Database was open successfully"
2020-05-23 11:21:34: "Checking for database schema update..."
2020-05-23 11:21:34: "Current database version", "29"
2020-05-23 11:21:34: "Initializing tables..."
2020-05-23 11:21:34: "PRAGMA table_info("folders")"
2020-05-23 11:21:34: "PRAGMA table_info("notes")"
2020-05-23 11:21:34: "PRAGMA table_info("tags")"
2020-05-23 11:21:34: "PRAGMA table_info("note_tags")"
2020-05-23 11:21:34: "PRAGMA table_info("sync_items")"
2020-05-23 11:21:34: "PRAGMA table_info("version")"
2020-05-23 11:21:34: "PRAGMA table_info("deleted_items")"
2020-05-23 11:21:34: "PRAGMA table_info("settings")"
2020-05-23 11:21:34: "PRAGMA table_info("alarms")"
2020-05-23 11:21:34: "PRAGMA table_info("master_keys")"
2020-05-23 11:21:34: "PRAGMA table_info("item_changes")"
2020-05-23 11:21:34: "PRAGMA table_info("note_resources")"
2020-05-23 11:21:34: "PRAGMA table_info("resource_local_states")"
2020-05-23 11:21:34: "PRAGMA table_info("resources")"
2020-05-23 11:21:34: "PRAGMA table_info("notes_normalized")"
2020-05-23 11:21:34: "PRAGMA table_info("revisions")"
2020-05-23 11:21:34: "PRAGMA table_info("migrations")"
2020-05-23 11:21:34: "PRAGMA table_info("resources_to_download")"
2020-05-23 11:21:34: "PRAGMA table_info("key_values")"
2020-05-23 11:21:34: "PRAGMA table_info("sqlite_stat1")"
```
So no difference in the log file.

Using again the original db file (1.0.214 ):
query: `SELECT * FROM sqlite_master;`
delivers as csv-output:
```
type,name,tbl_name,rootpage,sql
table,folders,folders,2,"CREATE TABLE folders (id TEXT PRIMARY KEY, title TEXT NOT NULL DEFAULT """", created_time INT NOT NULL, updated_time INT NOT NULL, user_created_time INT NOT NULL DEFAULT 0, user_updated_time INT NOT NULL DEFAULT 0, encryption_cipher_text TEXT NOT NULL DEFAULT """", encryption_applied INT NOT NULL DEFAULT 0, parent_id TEXT NOT NULL DEFAULT """", is_shared INT NOT NULL DEFAULT 0)"
index,sqlite_autoindex_folders_1,folders,3,NULL
index,folders_title,folders,4,CREATE INDEX folders_title ON folders (title)
index,folders_updated_time,folders,5,CREATE INDEX folders_updated_time ON folders (updated_time)
table,notes,notes,6,"CREATE TABLE notes (id TEXT PRIMARY KEY, parent_id TEXT NOT NULL DEFAULT """", title TEXT NOT NULL DEFAULT """", body TEXT NOT NULL DEFAULT """", created_time INT NOT NULL, updated_time INT NOT NULL, is_conflict INT NOT NULL DEFAULT 0, latitude NUMERIC NOT NULL DEFAULT 0, longitude NUMERIC NOT NULL DEFAULT 0, altitude NUMERIC NOT NULL DEFAULT 0, author TEXT NOT NULL DEFAULT """", source_url TEXT NOT NULL DEFAULT """", is_todo INT NOT NULL DEFAULT 0, todo_due INT NOT NULL DEFAULT 0, todo_completed INT NOT NULL DEFAULT 0, source TEXT NOT NULL DEFAULT """", source_application TEXT NOT NULL DEFAULT """", application_data TEXT NOT NULL DEFAULT """", `order` INT NOT NULL DEFAULT 0, user_created_time INT NOT NULL DEFAULT 0, user_updated_time INT NOT NULL DEFAULT 0, encryption_cipher_text TEXT NOT NULL DEFAULT """", encryption_applied INT NOT NULL DEFAULT 0, `markup_language` INT NOT NULL DEFAULT 1, is_shared INT NOT NULL DEFAULT 0)"
index,sqlite_autoindex_notes_1,notes,7,NULL
index,notes_title,notes,8,CREATE INDEX notes_title ON notes (title)
index,notes_updated_time,notes,9,CREATE INDEX notes_updated_time ON notes (updated_time)
index,notes_is_conflict,notes,10,CREATE INDEX notes_is_conflict ON notes (is_conflict)
index,notes_is_todo,notes,11,CREATE INDEX notes_is_todo ON notes (is_todo)
index,notes_order,notes,12,CREATE INDEX notes_order ON notes (`order`)
table,tags,tags,13,"CREATE TABLE tags (id TEXT PRIMARY KEY, title TEXT NOT NULL DEFAULT """", created_time INT NOT NULL, updated_time INT NOT NULL, user_created_time INT NOT NULL DEFAULT 0, user_updated_time INT NOT NULL DEFAULT 0, encryption_cipher_text TEXT NOT NULL DEFAULT """", encryption_applied INT NOT NULL DEFAULT 0, is_shared INT NOT NULL DEFAULT 0)"
index,sqlite_autoindex_tags_1,tags,14,NULL
index,tags_title,tags,15,CREATE INDEX tags_title ON tags (title)
index,tags_updated_time,tags,16,CREATE INDEX tags_updated_time ON tags (updated_time)
table,note_tags,note_tags,17,"CREATE TABLE note_tags (id TEXT PRIMARY KEY, note_id TEXT NOT NULL, tag_id TEXT NOT NULL, created_time INT NOT NULL, updated_time INT NOT NULL, user_created_time INT NOT NULL DEFAULT 0, user_updated_time INT NOT NULL DEFAULT 0, encryption_cipher_text TEXT NOT NULL DEFAULT """", encryption_applied INT NOT NULL DEFAULT 0, is_shared INT NOT NULL DEFAULT 0)"
index,sqlite_autoindex_note_tags_1,note_tags,18,NULL
index,note_tags_note_id,note_tags,19,CREATE INDEX note_tags_note_id ON note_tags (note_id)
index,note_tags_tag_id,note_tags,20,CREATE INDEX note_tags_tag_id ON note_tags (tag_id)
index,note_tags_updated_time,note_tags,21,CREATE INDEX note_tags_updated_time ON note_tags (updated_time)
table,table_fields,table_fields,28,"CREATE TABLE table_fields (id INTEGER PRIMARY KEY, table_name TEXT NOT NULL, field_name TEXT NOT NULL, field_type INT NOT NULL, field_default TEXT)"
table,sync_items,sync_items,29,"CREATE TABLE sync_items (id INTEGER PRIMARY KEY, sync_target INT NOT NULL, sync_time INT NOT NULL DEFAULT 0, item_type INT NOT NULL, item_id TEXT NOT NULL, sync_disabled INT NOT NULL DEFAULT ""0"", sync_disabled_reason TEXT NOT NULL DEFAULT """", force_sync INT NOT NULL DEFAULT 0, item_location INT NOT NULL DEFAULT 1)"
index,sync_items_sync_time,sync_items,30,CREATE INDEX sync_items_sync_time ON sync_items (sync_time)
index,sync_items_sync_target,sync_items,31,CREATE INDEX sync_items_sync_target ON sync_items (sync_target)
index,sync_items_item_type,sync_items,32,CREATE INDEX sync_items_item_type ON sync_items (item_type)
index,sync_items_item_id,sync_items,33,CREATE INDEX sync_items_item_id ON sync_items (item_id)
table,version,version,36,"CREATE TABLE version (version INT NOT NULL, table_fields_version INT NOT NULL DEFAULT 0)"
table,deleted_items,deleted_items,34,"CREATE TABLE deleted_items (id INTEGER PRIMARY KEY, item_type INT NOT NULL, item_id TEXT NOT NULL, deleted_time INT NOT NULL, sync_target INT NOT NULL)"
index,deleted_items_sync_target,deleted_items,38,CREATE INDEX deleted_items_sync_target ON deleted_items (sync_target)
table,settings,settings,26,"CREATE TABLE `settings`(`key` TEXT PRIMARY KEY,`value` TEXT)"
index,sqlite_autoindex_settings_1,settings,27,NULL
index,notes_user_updated_time,notes,39,CREATE INDEX notes_user_updated_time ON notes (user_updated_time)
index,folders_user_updated_time,folders,40,CREATE INDEX folders_user_updated_time ON folders (user_updated_time)
index,tags_user_updated_time,tags,41,CREATE INDEX tags_user_updated_time ON tags (user_updated_time)
index,note_tags_user_updated_time,note_tags,42,CREATE INDEX note_tags_user_updated_time ON note_tags (user_updated_time)
table,alarms,alarms,44,"CREATE TABLE alarms (id INTEGER PRIMARY KEY AUTOINCREMENT, note_id TEXT NOT NULL, trigger_time INT NOT NULL)"
table,sqlite_sequence,sqlite_sequence,45,"CREATE TABLE sqlite_sequence(name,seq)"
index,alarm_note_id,alarms,46,CREATE INDEX alarm_note_id ON alarms (note_id)
table,master_keys,master_keys,47,"CREATE TABLE master_keys (id TEXT PRIMARY KEY, created_time INT NOT NULL, updated_time INT NOT NULL, source_application TEXT NOT NULL, encryption_method INT NOT NULL, checksum TEXT NOT NULL, content TEXT NOT NULL)"
index,sqlite_autoindex_master_keys_1,master_keys,48,NULL
index,notes_encryption_applied,notes,49,CREATE INDEX notes_encryption_applied ON notes (encryption_applied)
index,folders_encryption_applied,folders,50,CREATE INDEX folders_encryption_applied ON folders (encryption_applied)
index,tags_encryption_applied,tags,51,CREATE INDEX tags_encryption_applied ON tags (encryption_applied)
index,note_tags_encryption_applied,note_tags,52,CREATE INDEX note_tags_encryption_applied ON note_tags (encryption_applied)
table,item_changes,item_changes,54,"CREATE TABLE item_changes (id INTEGER PRIMARY KEY AUTOINCREMENT, item_type INT NOT NULL, item_id TEXT NOT NULL, type INT NOT NULL, created_time INT NOT NULL, source INT NOT NULL DEFAULT 1, before_change_item TEXT NOT NULL DEFAULT """")"
index,item_changes_item_id,item_changes,55,CREATE INDEX item_changes_item_id ON item_changes (item_id)
index,item_changes_created_time,item_changes,56,CREATE INDEX item_changes_created_time ON item_changes (created_time)
index,item_changes_item_type,item_changes,58,CREATE INDEX item_changes_item_type ON item_changes (item_type)
table,note_resources,note_resources,59,"CREATE TABLE note_resources (id INTEGER PRIMARY KEY, note_id TEXT NOT NULL, resource_id TEXT NOT NULL, is_associated INT NOT NULL, last_seen_time INT NOT NULL)"
index,note_resources_note_id,note_resources,60,CREATE INDEX note_resources_note_id ON note_resources (note_id)
index,note_resources_resource_id,note_resources,57,CREATE INDEX note_resources_resource_id ON note_resources (resource_id)
table,resource_local_states,resource_local_states,120,"CREATE TABLE resource_local_states (id INTEGER PRIMARY KEY, resource_id TEXT NOT NULL, fetch_status INT NOT NULL DEFAULT ""2"", fetch_error TEXT NOT NULL DEFAULT """")"
index,resource_local_states_resource_id,resource_local_states,122,CREATE INDEX resource_local_states_resource_id ON resource_local_states (resource_id)
index,resource_local_states_resource_fetch_status,resource_local_states,123,CREATE INDEX resource_local_states_resource_fetch_status ON resource_local_states (fetch_status)
table,resources,resources,22,"CREATE TABLE `resources`(`id` TEXT PRIMARY KEY,`title` TEXT NOT NULL DEFAULT """",`mime` TEXT NOT NULL,`filename` TEXT NOT NULL DEFAULT """",`created_time` INT NOT NULL,`updated_time` INT NOT NULL,`user_created_time` INT NOT NULL DEFAULT 0,`user_updated_time` INT NOT NULL DEFAULT 0,`file_extension` TEXT NOT NULL DEFAULT """",`encryption_cipher_text` TEXT NOT NULL DEFAULT """",`encryption_applied` INT NOT NULL DEFAULT 0,`encryption_blob_encrypted` INT NOT NULL DEFAULT 0, `size` INT NOT NULL DEFAULT -1, is_shared INT NOT NULL DEFAULT 0)"
index,sqlite_autoindex_resources_1,resources,23,NULL
table,notes_normalized,notes_normalized,131,"CREATE TABLE notes_normalized (id TEXT NOT NULL, title TEXT NOT NULL DEFAULT """", body TEXT NOT NULL DEFAULT """")"
index,notes_normalized_id,notes_normalized,146,CREATE INDEX notes_normalized_id ON notes_normalized (id)
table,notes_fts,notes_fts,0,"CREATE VIRTUAL TABLE notes_fts USING fts4(content=""notes_normalized"", notindexed=""id"", id, title, body)"
table,notes_fts_segments,notes_fts_segments,64,"CREATE TABLE 'notes_fts_segments'(blockid INTEGER PRIMARY KEY, block BLOB)"
table,notes_fts_segdir,notes_fts_segdir,109,"CREATE TABLE 'notes_fts_segdir'(level INTEGER,idx INTEGER,start_block INTEGER,leaves_end_block INTEGER,end_block INTEGER,root BLOB,PRIMARY KEY(level, idx))"
index,sqlite_autoindex_notes_fts_segdir_1,notes_fts_segdir,111,NULL
table,notes_fts_docsize,notes_fts_docsize,132,"CREATE TABLE 'notes_fts_docsize'(docid INTEGER PRIMARY KEY, size BLOB)"
table,notes_fts_stat,notes_fts_stat,135,"CREATE TABLE 'notes_fts_stat'(id INTEGER PRIMARY KEY, value BLOB)"
trigger,notes_fts_before_update,notes_normalized,0,CREATE TRIGGER notes_fts_before_update BEFORE UPDATE ON notes_normalized BEGIN\n						DELETE FROM notes_fts WHERE docid=old.rowid;\n					END
trigger,notes_fts_before_delete,notes_normalized,0,CREATE TRIGGER notes_fts_before_delete BEFORE DELETE ON notes_normalized BEGIN\n						DELETE FROM notes_fts WHERE docid=old.rowid;\n					END
trigger,notes_after_update,notes_normalized,0,"CREATE TRIGGER notes_after_update AFTER UPDATE ON notes_normalized BEGIN\n						INSERT INTO notes_fts(docid, id, title, body) SELECT rowid, id, title, body FROM notes_normalized WHERE new.rowid = notes_normalized.rowid;\n					END"
trigger,notes_after_insert,notes_normalized,0,"CREATE TRIGGER notes_after_insert AFTER INSERT ON notes_normalized BEGIN\n						INSERT INTO notes_fts(docid, id, title, body) SELECT rowid, id, title, body FROM notes_normalized WHERE new.rowid = notes_normalized.rowid;\n					END"
table,revisions,revisions,155,"CREATE TABLE revisions (id TEXT PRIMARY KEY, parent_id TEXT NOT NULL DEFAULT """", item_type INT NOT NULL, item_id TEXT NOT NULL, item_updated_time INT NOT NULL, title_diff TEXT NOT NULL DEFAULT """", body_diff TEXT NOT NULL DEFAULT """", metadata_diff TEXT NOT NULL DEFAULT """", encryption_cipher_text TEXT NOT NULL DEFAULT """", encryption_applied INT NOT NULL DEFAULT 0, updated_time INT NOT NULL, created_time INT NOT NULL)"
index,sqlite_autoindex_revisions_1,revisions,157,NULL
index,revisions_parent_id,revisions,172,CREATE INDEX revisions_parent_id ON revisions (parent_id)
index,revisions_item_type,revisions,189,CREATE INDEX revisions_item_type ON revisions (item_type)
index,revisions_item_id,revisions,212,CREATE INDEX revisions_item_id ON revisions (item_id)
index,revisions_item_updated_time,revisions,215,CREATE INDEX revisions_item_updated_time ON revisions (item_updated_time)
index,revisions_updated_time,revisions,224,CREATE INDEX revisions_updated_time ON revisions (updated_time)
table,migrations,migrations,238,"CREATE TABLE migrations (id INTEGER PRIMARY KEY, number INTEGER NOT NULL, updated_time INT NOT NULL, created_time INT NOT NULL)"
table,resources_to_download,resources_to_download,160,"CREATE TABLE resources_to_download (id INTEGER PRIMARY KEY, resource_id TEXT NOT NULL, updated_time INT NOT NULL, created_time INT NOT NULL)"
index,resources_to_download_resource_id,resources_to_download,180,CREATE INDEX resources_to_download_resource_id ON resources_to_download (resource_id)
index,resources_to_download_updated_time,resources_to_download,230,CREATE INDEX resources_to_download_updated_time ON resources_to_download (updated_time)
table,key_values,key_values,397,"CREATE TABLE key_values (id INTEGER PRIMARY KEY, `key` TEXT NOT NULL, `value` TEXT NOT NULL, `type` INT NOT NULL, updated_time INT NOT NULL)"
index,key_values_key,key_values,399,CREATE UNIQUE INDEX key_values_key ON key_values (key)
view,tags_with_note_count,tags_with_note_count,0,"CREATE VIEW tags_with_note_count AS \n						SELECT tags.id as id, tags.title as title, tags.created_time as created_time, tags.updated_time as updated_time, COUNT(notes.id) as note_count \n						FROM tags \n							LEFT JOIN note_tags nt on nt.tag_id = tags.id \n							LEFT JOIN notes on notes.id = nt.note_id \n						WHERE notes.id IS NOT NULL \n						GROUP BY tags.id"
index,resources_size,resources,514,CREATE INDEX resources_size ON resources(size)
table,sqlite_stat1,sqlite_stat1,429,"CREATE TABLE sqlite_stat1(tbl,idx,stat)"
table,sqlite_stat4,sqlite_stat4,430,"CREATE TABLE sqlite_stat4(tbl,idx,neq,nlt,ndlt,sample)"
```
hopefully you can detect the bug. thanks.

**Comment by laurent22** [comment]
Did you previously open the sqlite file with some other tool? I see sqlite_stat1 and sqlite_stat4 tables which normally shouldn't be there, and I don't think normal usage of the app would create them. If you delete them it should work again, and I'm also going to add an exception for tables named like this.

**Comment by amatter** [comment]
Actually I did now a (tool used : https://fishcodelib.com/Database.htm (which I like a lot btw))
```
drop table sqlite_stat1;
drop table sqlite_stat4;
```
which did solve the problem. Thank you a lot!

I'll try now to upgrade my other instances as well over the weekend.

Remark: It might well be that I did open the sqlite database earlier with another tool. But for sure not after using joplin 1.0.201 or before the upgrade to 1.0.214 .
